core/src/main/java/org/apache/calcite/sql/SqlSplittableAggFunction.java

```diff
@@ -268,6 +303,19 @@ public AggregateCall topSplit(RexBuilder rexBuilder,
           aggregateCall.type, aggregateCall.name);
     }
 
+    public AggregateCall merge(AggregateCall top, AggregateCall bottom) {
+      SqlKind topKind = top.getAggregation().getKind();
+      if (topKind == bottom.getAggregation().getKind()
+          && (topKind == SqlKind.SUM
+              || topKind == SqlKind.SUM0)) {
+        return AggregateCall.create(bottom.getAggregation(), bottom.isDistinct(),
+            bottom.isApproximate(), bottom.getArgList(), bottom.filterArg,
+            bottom.getCollation(), bottom.getType(), top.getName());
+      } else {
+        return null;
+      }
+    }
+
```

Comment:
Another test case where both top and bottom aggregate has empty group by and are mergeable (e.g. SUM of SUM) would be nice.